test_crates/pub_function_changed_abi/new/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+pub extern "Rust" fn c_function_becomes_rust() -> () {}
+
+pub fn c_function_becomes_rust_implicitly() -> () {}
+
+pub extern "C" fn rust_function_becomes_c() -> () {}
+
+pub extern "C" fn implicit_rust_function_becomes_c() -> () {}
+
+pub extern "Rust" fn implicit_c_function_becomes_rust() -> () {}
+
+pub extern "Rust" fn rust_function_remains_rust() -> () {}
```

Comment:
Suggestion for a couple more cases here:
- an explicitly-Rust ABI function remains Rust ABI but implicitly so, by removing the `extern "Rust"`
- the same thing in the opposite direction: adding explicit Rust ABI to an already implicitly-Rust ABI function

These cases are important guards against possible _false-positives_, which are very annoying for users and so are at least as important to protect against as the test cases for true-positives that make sure our functionality continues to work.